examples/flow-counter/containers/App.js

```diff
@@ -0,0 +1,33 @@
+/* @flow */
+
+import { connect } from 'react-redux';
+import Counter from '../components/Counter';
+import { increment, decrement, incrementIfOdd, incrementAsync } from '../actions/counter';
+import type { AppState, Dispatchable } from '../types';
+import type { Dispatch } from 'redux';
+
+type StateProps = AppState;
+
+function mapStateToProps(state : AppState) : StateProps {
+  return {
+    counter: state.counter
+  };
+}
+
+type DispatchProps = {
+  increment: () => any,
+  decrement: () => any,
+  incrementIfOdd: () => any,
+  incrementAsync: () => any
+}
+
+function mapDispatchToProps(dispatch: Dispatch<Dispatchable>) : DispatchProps {
+  return {
+    increment: () => dispatch(increment()),
+    decrement: () => dispatch(decrement()),
+    incrementIfOdd: () => dispatch(incrementIfOdd()),
+    incrementAsync: () => dispatch(incrementAsync()),
+  };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(Counter);
```

Comment:
Finally made this to type check!!! The `$Diff` type was working, the problem was that the correct flow type for classes that extend `React.Component` is `ReactClass`, so that it knows how to infer the polymorphic props and state types